app/controllers/api/v1/latest_photos_controller.rb

```diff
@@ -1,26 +1,21 @@
 class Api::V1::LatestPhotosController < ApplicationController
   def index
-    def index
-      @rover = Rover.find_by name: params[:rover_id].titleize
-      if @rover
-        render json: search_photos, each_serializer: PhotoSerializer, root: :latest_photos
-      else
-        render json: { errors: "Invalid Rover Name" }, status: :bad_request
-      end
-    end
-  end
+    rover = Rover.find_by name: params[:rover_id].titleize
 
-  private
+    if rover
+      validated_params = params
+        .permit(:rover_id, :camera, :earth_date, :size, :page, :per_page)
+        .merge(sol: rover.photos.maximum(:sol))
+      photos = helpers.search_photos rover, validated_params
```

Comment:
If you add `include PhotoHelper` to the top of the controller, you can remove `helpers.` and also the `PhotoHelper::` namespace of the InvalidSizeParameter.

Reply:
I read [this warning](https://stackoverflow.com/a/11161692) about using the `include` keyword. Is this something we are concerned with?

Reply:
Hm, I don't think it's relevant, because if the `config/routes.rb` file is configured correctly there won't be URLs that will reach any potential extra controller actions.